File: examples_android/app/src/main/java/com/confirmit/testsdkapp/layouts/defaultlayoutjava/items/DefaultLayoutTextBox.java
package com.confirmit.testsdkapp.layouts.defaultlayoutjava.items;

public class DefaultLayoutTextBox implements DefaultLayoutItem {
    private DefaultLayoutTextBoxType textBoxType;
    private String initialValue;

    DefaultLayoutTextBox(DefaultLayoutTextBoxType textBoxType, String initialValue) {
        this.textBoxType = textBoxType;
        this.initialValue = initialValue;
    }

    public DefaultLayoutTextBoxType getTextBoxType() {
        return textBoxType;
    }

    public String getInitialValue() {
        return initialValue;
    }
}

File: examples_ios/MyApp/Sources/SurveyLayouts/DefaultLayout/Items/DefaultLayoutTextBox.swift
import Foundation

enum DefaultLayoutTextBoxType {
    case text
    case numeric
}

class DefaultLayoutTextBox: DefaultLayoutItem {
    let textBoxType: DefaultLayoutTextBoxType
    let initialValue: String

    init(type: DefaultLayoutTextBoxType, initialValue: String) {
        self.textBoxType = type
        self.initialValue = initialValue
    }
}
